src/native/libs/System.IO.Compression.Native/extra_libs.cmake

```
@@ -13,7 +13,7 @@ macro(append_extra_compression_libs NativeLibsExtra)
       find_package(ZLIB REQUIRED)
       list(APPEND ZLIB_LIBRARIES m)
   else()
-    list(APPEND ZLIB_LIBRARIES zlib)
+    list(APPEND ZLIB_LIBRARIES z)
```

According to the PR description, you're forcing CLR_CMAKE_USE_SYSTEM_ZLIB to true, which means you want the zlib package installed in the system to be used. Why would you want that? The intention here is to consume the in-tree zlib that we built.

---

The goal of `CLR_CMAKE_USE_SYSTEM_ZLIB`  (and other similar `CLR_CMAKE_USE_SYSTEM_*` options) is to use the system's version of a library. There can be many reasons for that:

- The system has a newer/better version of the library. For example, with .NET 8 + Fedora 40, we are using zlib-ng already because that's the system version of zlib in Fedora: https://fedoraproject.org/wiki/Changes/ZlibNGTransition. So our users already benefit from zlib-ng by using the system version of zlib.
- Some distributions have a standard policy against using bundled libraries. For Fedora's stance, see https://fedoraproject.org/wiki/Bundled_Libraries
- Bundling libraries (such as zlib) means any security issue that's discovered in the library needs to be fixed in all places where the library is duplicated. This also applies to feature improvements and bugfixes but is more critical for addressing security fixes.

Backing up a bit, is your concern with the specific fix proposed in the PR or with `CLR_CMAKE_USE_SYSTEM_*`  itself?

---

> * The system has a newer/better version of the library. For example, with .NET 8 + Fedora 40, we are using zlib-ng already because that's the system version of zlib in Fedora

We are aiming to have runtime consume the latest version of zlib-ng as well. Our plan is to get immediate notifications when there's a new release of zlib-ng and update the in-tree code in our repo ASAP.

> * Some distributions have a standard policy against using bundled libraries. For Fedora's stance, see https://fedoraproject.org/wiki/Bundled_Libraries

@jkotas @jkoritzinsky this is new to me. Can we talk about this?

> is your concern with the specific fix proposed in the PR or with `CLR_CMAKE_USE_SYSTEM_*` itself?

My concern is with the change proposed in this PR because you would now be ignoring the in-tree zlib-ng for all platforms that should be consuming it, not just Fedora. In other words, _no one_ would be using the in-tree zlib-ng anymore. So this is not the right approach.

---

It's safer to set it conditionally:
```suggestion
    list(APPEND ZLIB_LIBRARIES $<IF:$<BOOL:CLR_CMAKE_USE_SYSTEM_ZLIB>,z,zlib>)
```

---

Yes, that would work!

---

It's an impactful change so I'll trigger a bunch of azp runs.

---

> @jkotas @jkoritzinsky this is new to me. Can we talk about this?

I had a brief conversation with @richlander and @jkoritzinsky about this so the suggestion applied above would solve the problem.

---

> In other words, no one would be using the in-tree zlib-ng anymore. So this is not the right approach.

Aah. Sorry about that. That was not my intention.

Thanks, @am11 for fixing this up!